src/types/common.d.ts

```diff
@@ -109,3 +109,25 @@ export interface ModifierContextState {
   text_line: number;
   top_text_line: number;
 }
+
+
+/* TODO: Move to EasyScore / Grammar / Parser */
+export type Match = string | Match[] | null;
+export interface Rule {
+  // Lexer Rules
+  token?: string; // The token property is a string that is compiled into a RegExp.
+  noSpace?: boolean; // TODO: None of the EasyScore rules specify noSpace, so it is not used anywhere.
+
+  // Parser Rules
+  expect?: RuleFunction[];
+  zeroOrMore?: boolean;
+  oneOrMore?: boolean;
+  maybe?: boolean;
+  or?: boolean;
+  run?: TriggerFunction;
+}
+export interface TriggerState {
```

Comment:
This can be inlined in `TriggerFunction`

Reply:
(i.e., no need for a separate named interface `TriggerState`.

Reply:
Sounds good. In the future if we make the TriggerState object more complicated, Parser might have to export it as a named interface, so that external classes (i.e. EasyScore) can import the type. For now it's a simple struct, so inlining makes sense.